### PR TITLE
RATEST-186:Replace headerPage.clickOnHomeIcon(); with HomePage().go() and Remove HeaderPage class 

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-186 
Have removed header page in the system because it have elements that are not visible hence making the system to take long,

Another refactoring was to make the system use  homepage.go since it's the only functionality being used in homePage.

All the tests are passing well after testing with care

### CODE REVIEW

**Comment by sherrif10** [comment]
cc @kdaud  waiting for ur approval

**Comment by kdaud** [comment]
Was the presence of header page and its functionality failing any test?

**Comment by sherrif10** [comment]
HeaderPage functionalities doesnot make any sense to me at all, Most of the elements in header Page forexample  By.cssSelector(".logout > a"); used to bring un burning errors on . ci , clickOnHomeIcon, this as well used to throw clickable exceptions not until i used HomePage.go() which acts the same and makes the system speedy, Hope i eraborated it cleary, Let me know if some clarifications are needed

**Comment by sherrif10** [comment]
cc @kdaud  any more suggestions.

**Comment by kdaud** [comment]
@sherrif10 was the presence of header page and its functionality failing any test?

**Comment by sherrif10** [comment]
> was the presence of header page and its functionality failing any test?

It was not failing any test. thanks

**Comment by sherrif10** [comment]
cc @kdaud , this ticket was created to simplify the code workflow and make the test run smoothly, Any more suggestion or approval

**Comment by kdaud** [comment]
>  make the test run smoothly

@sherrif10 which test was not running smoothly before?

**Comment by sherrif10** [comment]
You can use homePage.clickOnHomeIcon() on each test i have changed and compare the speed at which the server runs, FormTest was one of the test failing.

File: ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:81]: Worth looking into the indention

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:79]: Even here

File: ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:81]: Identation is the same here try to view the file using github view file

File: ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/UsingBackButtonInMergePatientTest.java:79]: Identation is the correct one

**Comment by kdaud** [comment]
@sherrif10 worth renaming the commit message to what actually the PR is doing!

**Comment by sherrif10** [comment]
> worth renaming the commit message to what actually the PR is doing!

Pull request is getting rid of headerPage and its functionalities, that's what the changes are doing as well.

**Comment by kdaud** [comment]
> Pull request is getting rid of headerPage and its functionalities

@sherrif10 it would be better to be reflected in the commit message.

**Comment by sherrif10** [comment]
> it would be better to be reflected in the commit message.

What is not exactly reflected in the commit message , The commit message is saying Replace HeaderPage.clickOnHomeIcon with HeaderPagePage.go() , is this not worth changes reflected in this commit!

**Comment by kdaud** [comment]
The changes are ensuring header page and its functionality is omitted to rather use home page functionality instead.

**Comment by sherrif10** [comment]
@kdaud  check the description in the commit message . Have added the description of what this ticket is handling

**Comment by kdaud** [comment]
> check description in the commit message

Would be better to be captured in the commit message.

**Comment by sherrif10** [comment]
> Would be better to be captured in the commit message.

Please we cannot write the whole description as the  commit message. One of the main advantage of commit messages is to show direct reference to the changes we tend to accomplish, Am not sure how you would write that as the  commit message

**Review by kdaud — APPROVED** [review]
cc: @dkayiwa

**Pull request merged**
